### ISSUE TITLE
Possible memory leak when initialising child element via constructor

### ISSUE THREAD

**Issue by vedtam** [Polymer/lit-element#689]
Hi,

While initialising a child element as bellow, it seems like a memory gets created:

```
import { LitElement, html, css, property } from 'lit-element';
import { TextInput } from './text-input';

export class SearchAutocomplete extends LitElement {
  searchInput: TextInput = new TextInput(); // <-- creates memory leak
  ...
```

Chrome isn't able to GC my web-app, so after 4-5 page reload I end up with around ~300MB of memory that's stuck and can't even manually be cleared (by clicking the bin - top left or switching tabs) but via closing and reopening the web-app in a new tab.

If I create a plain class:
```
export class TextInputTest {
  sayHello() {
    console.log('hello world');
  }
}
```

than use it for the initialisation, the leak is gone:
```
export class SearchAutocomplete extends LitElement {
  searchInput: TextInputTest = new TextInputTest();
```

git repo to demo:
https://github.com/vedtam/lit-leak-test

I have created a demo that can be tested on stackblitz as well, but for most accurate testing the code from the repo would be more appropriate:
https://stackblitz.com/edit/lit-element-leak-demo?file=app-main.ts

Regards,
vedtam.

**Comment by anoblet** [comment]
I'm able to replicate this on Windows/Chrome 74

**Comment by sorvell** [comment]
Thanks for the repro. A memory leak across page refreshes would be a browser related bug. It's unclear right now if this is actually a leak. I modified your index.html to include the following to be able to check memory with the Chrome dev tools open. Careful as if there's a leak, this will eventually crash the browser:

```
<script>
    document.title = new Intl.NumberFormat().format(performance.memory.totalJSHeapSize);
    window.location.reload();
  </script>
```

When I run your app with this addition, I see the usage never go above about 350MB, cycling down after that. With the dev tools open, however, the number keeps going up.

I'll try to check with a v8 expert to see if this is an issue that should be filed. It does seem, at least, like there's a Chrome dev tools issue.

**Comment by sorvell** [comment]
Filed as https://bugs.chromium.org/p/chromium/issues/detail?id=968603.

**Comment by vedtam** [comment]
Hi,

This becomes more and more interesting, I haven't tried with dev tools closed. Will follow closely the issue you filed on chrome.

**Comment by kylerberry** [comment]
not sure if related or helpful here, but I was experiencing a similar issue across page refreshes. My use case had an async/await resolving inside a reoccurring animationFrame request. I ended up boiling my example down until there was no more lit-element and the problem persisted until I closed and reopened the tab.

Seems to validate the issue with progressive memory leak from chrome tab process.
